src/gravel/controllers/nodes/local.py

```diff
@@ -0,0 +1,185 @@
+# project aquarium's backend
+# Copyright (C) 2021 SUSE, LLC.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+import asyncio
+from enum import Enum
+from typing import List
+
+import psutil
+from pydantic import BaseModel, Field
+
+# Minimum requirements for individual host validation:
+# NOTE(jhesketh): These are obviously hardcoded for the moment, but may be
+#                 better moved into some kind of support matrix (eg feature
+#                 vs hardware etc).
+#                 These are also currently somewhat arbitrary and will need
+#                 tuning.
+AQUARIUM_MIN_CPU = 2
+AQUARIUM_MIN_MEMORY_GB = 2
+AQUARIUM_MIN_ROOT_DISK_GB = 10
+
+
+class HardwareQualifiedEnum(int, Enum):
+    QUALIFIED = 0
+    ERROR = 1
+
+
+class CPUQualifiedModel(BaseModel):
+    qualified: bool = Field("The CPU is sufficient")
+    min_cpu: int = Field("Minimum number of CPU's")
```

Comment:
I'm wondering if it's possible/sensible to use just `min` instead of `min_cpu` and `actual` for `actual_cpu` (same for `min_mem`, `actual_mem`, `min_disk`, `actual_disk`)? That way the entries would be more generic. Or is there an actual reason against it?

Reply:
The main reason is that we may validate more about a piece. For example, we may have `min_cores`, `min_clock` on the CPU and `min_speed`, `min_mem` on the memory. I should update `min_cpu` to be `min_threads` since that is what we are checking.

Reply:
I see, thanks :+1: